src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -175,8 +175,10 @@ internal static void AddRazorViewEngineServices(IServiceCollection services)
             // This caches Razor page activation details that are valid for the lifetime of the application.
             services.TryAddSingleton<IRazorPageActivator, RazorPageActivator>();
 
-            // Only want one ITagHelperActivator so it can cache Type activation information. Types won't conflict.
+            // Only want one ITagHelperActivator and ITagHelperComponentContextInitializer so it can cache Type activation information. Types won't conflict.
             services.TryAddSingleton<ITagHelperActivator, DefaultTagHelperActivator>();
+            services.TryAddSingleton<ITagHelperComponentContextInitializer, DefaultTagHelperComponentContextInitializer>();
+
```

Comment:
Not sure if this needs to be scoped to match the TagHelperComponentManager

Reply:
Has no request state and is therefore fine as-is